test/models/PackageSpec.scala

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models
+
+import generators.ModelGenerators
+import org.scalacheck.Gen
+import org.scalatest.{FreeSpec, MustMatchers}
+import org.scalatestplus.scalacheck.ScalaCheckPropertyChecks
+import play.api.libs.json.{JsSuccess, Json}
+
+class PackageSpec extends FreeSpec with MustMatchers with ScalaCheckPropertyChecks with ModelGenerators {
+
+  "Package" - {
+
+    "must deserialise to a Bulk Package" in {
+
+      forAll(Gen.oneOf(BulkPackage.validCodes)) {
+        kindOfPackage =>
+
+          val json = Json.obj("kindOfPackage" -> kindOfPackage)
+
+          val expectedPackage = BulkPackage(kindOfPackage, None)
+
+          json.validate[Package] mustEqual JsSuccess(expectedPackage)
+      }
+    }
+
+    "must deserialise to an Unpacked Package" in {
+
+      val gen = for {
+        kindOfPackage   <- Gen.oneOf(UnpackedPackage.validCodes)
+        numberOfPieces  <- Gen.choose(1, 99999)
+      } yield (kindOfPackage, numberOfPieces)
+
+      forAll(gen) {
+        case (kindOfPackage, numberOfPieces) =>
+
+          val json = Json.obj(
+            "kindOfPackage"   -> kindOfPackage,
+            "numberOfPieces"  -> numberOfPieces
+          )
+
+          val expectedPackage = UnpackedPackage(kindOfPackage, numberOfPieces, None)
+
+          json.validate[Package] mustEqual JsSuccess(expectedPackage)
+      }
+    }
+
+    "must deserialise to a Regular Package" in {
+
+      val gen = for {
+        kindOfPackage    <- stringWithMaxLength(3)
```

Comment:
Would changing this to `Gen.oneOf(BulkPackage.validCodes)` mean we can remove the `whenever` below?

Reply:
This test is for a `RegularPackage`, so it needs to have any `kindOfPackage` that _isn't_ a valid bulk or unpacked code.  Short of specifying all valid "regular" codes I don't think we can avoid the whenever